From 9ba57e09ab7cc419f3a70e39e58e573112c00714 Mon Sep 17 00:00:00 2001
From: Izzul Irfan <muhammadx.izzul.irfan.bin.che.iswanizam@intel.com>
Date: Tue, 6 Jan 2026 09:46:25 +0800
Subject: [PATCH] media: platform: intel: Add AR0820 sensor support to IPU ACPI
 device table

Register the AR0820 sensor in the supported_devices array
when CONFIG_VIDEO_AR0820 is enabled.
Add the AR0820 ACPI HID to the ipu_acpi_match table
for proper device detection.
Enables platform integration and detection of
the SENSING AR0820 sensor via ACPI.

Signed-off-by: Tay, Boon Wooi <boonx.wooi.tay@intel.com>
Signed-off-by: Izzul Irfan <muhammadx.izzul.irfan.bin.che.iswanizam@intel.com>
---
 Makefile                                |  4 ++++
 drivers/media/platform/intel/ipu-acpi.c |  7 +++++++
 include/media/i2c/ar0820.h              | 24 ++++++++++++++++++++++++
 include/media/serdes-pdata.h            |  4 ++++
 4 files changed, 39 insertions(+)
 create mode 100644 include/media/i2c/ar0820.h

diff --git a/Makefile b/Makefile
index ac0466c26..96b1b8ddf 100644
--- a/Makefile
+++ b/Makefile
@@ -46,6 +46,7 @@ export CONFIG_VIDEO_AR0234 = m
 # kernel version >= 6.12.15
 export CONFIG_VIDEO_ISX031=m
 export CONFIG_VIDEO_MAX9X=m
+export CONFIG_VIDEO_AR0820=m
 
 obj-y += drivers/media/i2c/
 
@@ -75,6 +76,9 @@ subdir-ccflags-$(CONFIG_INTEL_SKL_INT3472) += \
 subdir-ccflags-$(CONFIG_VIDEO_ISX031) += \
 	-DCONFIG_VIDEO_ISX031
 
+subdir-ccflags-$(CONFIG_VIDEO_AR0820) += \
+    -DCONFIG_VIDEO_AR0820
+
 subdir-ccflags-y += $(subdir-ccflags-m)
 
 all:
diff --git a/drivers/media/platform/intel/ipu-acpi.c b/drivers/media/platform/intel/ipu-acpi.c
index 8b94dc307..ccb331852 100644
--- a/drivers/media/platform/intel/ipu-acpi.c
+++ b/drivers/media/platform/intel/ipu-acpi.c
@@ -63,6 +63,10 @@ static const struct ipu_acpi_devices supported_devices[] = {
 	{ "INTC031M", ISX031_NAME, get_sensor_pdata, NULL, 0, TYPE_SERDES, "max9x",
 		ISX031_I2C_ADDRESS, 1600 },	// D3 ISX031 HID
 #endif
+#if IS_ENABLED(CONFIG_VIDEO_AR0820)
+	{ "AR0820", AR0820_NAME, get_sensor_pdata, NULL, 0, TYPE_SERDES, "max9x",
+		AR0820_I2C_ADDRESS, 1600 },  // SENSING AR0820 HID
+#endif
 #endif
 };
 
@@ -88,6 +92,9 @@ static const struct acpi_device_id ipu_acpi_match[] = {
 #if IS_ENABLED(CONFIG_VIDEO_ISX031)
 	{ "INTC1031", 0 },	// ISX031 HID
 	{ "INTC031M", 0 },	// D3CMC68N-115-084 ISX031 HID
+#endif
+#if IS_ENABLED(CONFIG_VIDEO_AR0820)
+	{ "AR0820", 0 },    // AR0820 HID
 #endif
 	{},
 };
diff --git a/include/media/i2c/ar0820.h b/include/media/i2c/ar0820.h
new file mode 100644
index 000000000..b1e5949cd
--- /dev/null
+++ b/include/media/i2c/ar0820.h
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (C) 2014 - 2022 Intel Corporation */
+
+#ifndef __AR0820_H
+#define __AR0820_H
+
+#include <linux/types.h>
+
+#define AR0820_NAME "ar0820"
+
+#define AR0820_I2C_ADDRESS 0x6D	// Sensing ISP I2C Address is 0xDA >> 1
+
+struct ar0820_platform_data {
+        unsigned int port;
+        unsigned int lanes;
+        uint32_t i2c_slave_address;
+        int irq_pin;
+        unsigned int irq_pin_flags;
+        char irq_pin_name[16];
+        char suffix;
+        int gpios[4];
+};
+
+#endif /* __AR0820_H  */
diff --git a/include/media/serdes-pdata.h b/include/media/serdes-pdata.h
index 829f73014..63acfbe21 100644
--- a/include/media/serdes-pdata.h
+++ b/include/media/serdes-pdata.h
@@ -8,6 +8,10 @@
 #include <media/i2c/isx031.h>
 #endif
 
+#if IS_ENABLED(CONFIG_VIDEO_AR0820)
+#include <media/i2c/ar0820.h>
+#endif
+
 struct serdes_subdev_info {
 	struct i2c_board_info board_info;
 	int i2c_adapter_id;
-- 
2.43.0

